src/Microsoft.ML.Data/Depricated/Vector/VBufferMathUtils.cs

```diff
@@ -174,14 +174,14 @@ public static void AddMult(ref VBuffer<Float> src, Float c, ref VBuffer<Float> d
                 return;
             }
             // REVIEW: Should we use SSE for any of these possibilities?
-            VBufferUtils.ApplyWith(ref src, ref dst, (int i, Float v1, ref Float v2) => v2 += c * v1);
+            VBufferUtils.ApplyWith(in src, ref dst, (int i, Float v1, ref Float v2) => v2 += c * v1);
         }
 
         /// <summary>
         /// Perform scalar vector addition
         /// <c><paramref name="res"/> = <paramref name="c"/> * <paramref name="src"/> + <paramref name="dst"/></c>
         /// </summary>
-        public static void AddMult(ref VBuffer<Float> src, Float c, ref VBuffer<Float> dst, ref VBuffer<Float> res)
+        public static void AddMult(in VBuffer<Float> src, Float c, ref VBuffer<Float> dst, ref VBuffer<Float> res)
```

Comment:
>ref [](start = 67, length = 3)

in

Reply:
`dst` can't be marked `in` because of the case that calls `VBufferUtils.ApplyWithCopy`, which passes `dst` into the `ref dst` parameter.

---
In reply to: [229499582](https://github.com/dotnet/machinelearning/pull/1454#discussion_r229499582) [](ancestors = 229499582)